typing/typecore.ml

```diff
@@ -10645,6 +10651,16 @@ let report_error ~loc env = function
         "@[Function arguments and returns must be representable.@]@ %a"
         (Jkind.Violation.report_with_offender
            ~offender:(fun ppf -> Printtyp.type_expr ppf ty)) violation
+  | Record_projection_not_rep (ty,violation) ->
+      Location.errorf ~loc
+        "@[Record projections must be representable.@]@ %a"
```

Comment:
I think it's not exactly that the projection is unrepresentable, but rather that the record being projected from is unrepresentable.